adhocteam/Head-Start-TTADP
# Request Permissions view & HTTPError handler

**Description of change**

Users who have an account but are denied access because they have no permissions should see a message helping them request permission. On the backend, `authMiddleware` will return a '403: Forbidden' http response, so the frontend should understand this as a permissions issue (different from a '401: Unauthorized').

This PR adds an `HTTPError` custom class, which is used by the frontend to handle HTTP error codes. The `App.js` will set `authError` if an HTTP 403 error is thrown when fetching a user.

**How to test**

Set `CURRENT_USER_ID=2` in your .env file (and restart the backend): Dumbledore is a user with no permissions, so you should see the "Request Permissions" page.

Create a user with only `SITE_ACCESS`, log in as that user and you should NOT see this message. Same for users with multiple permissions (including `SITE_ACCESS`).

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/139

**Checklist**

    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
